vm/src/tests/run_deprecated_contract_class_simplified.rs

```diff
@@ -0,0 +1,419 @@
+/* This file contains a test that runs the program: cairo_programs/starknet_os_deprecated_cc.cairo
+   For testsing purposes, the contract ran by this program is hardcoded, with values taken from compiling:
+
+   %lang starknet
+
+   @view
+   func get_number() -> (number: felt) {
+       let number = 14;
+       %{print("hello world")%}
+       return (number=number);
+   }
+
+   The purpose of this test is to check the functionality of the HintProcessor::execute_hint_extensive functionality
+   And to show a very simplified example on how it can be used to achieve the `vm_load_data` functionality used by starknet os programs
+*/
+use crate::stdlib::{collections::HashMap, prelude::*};
+
+use felt::{felt_str, Felt252};
+use num_traits::{Num, One, Zero};
+#[cfg(target_arch = "wasm32")]
+use wasm_bindgen_test::*;
+
+use crate::{
+    cairo_run::{cairo_run, CairoRunConfig},
+    hint_processor::{
+        builtin_hint_processor::{
+            builtin_hint_processor_definition::{BuiltinHintProcessor, HintProcessorData},
+            hint_utils::{get_ptr_from_var_name, insert_value_from_var_name},
+        },
+        hint_processor_definition::{
+            HintExtension, HintProcessor, HintProcessorLogic, HintReference,
+        },
+    },
+    serde::deserialize_program::ApTracking,
+    vm::{
+        errors::hint_errors::HintError,
+        runners::cairo_runner::{ResourceTracker, RunResources},
+        trace::trace_entry::RelocatedTraceEntry,
+        vm_core::VirtualMachine,
+    },
+};
+
+struct SimplifiedOsHintProcessor {
+    builtin_hint_processor: BuiltinHintProcessor,
+    run_resources: RunResources,
+}
+
+impl ResourceTracker for SimplifiedOsHintProcessor {
```

Comment:
I write this comment just to remember,
We should document this trait since developers are going to implement it